### PR TITLE
Drop php 5

### PR DESCRIPTION
**Pull request by greg0ire**
I am targeting this branch, because this is a major BC-break.

## Changelog

```markdown
### Removed
- PHP < 7.1 compatibility
```

Blocked by #142

### CODE REVIEW

**Comment by greg0ire** [comment]
@Soullivaneuh how can we allow the `::class` notation for this branch?

File: src/Writer/CsvWriter.php [hunk 1]
```diff
@@ -66,7 +66,7 @@ class CsvWriter implements TypedWriterInterface
      * @param bool   $showHeaders
      * @param bool   $withBom
      */
-    public function __construct($filename, $delimiter = ',', $enclosure = '"', $escape = '\\', $showHeaders = true, $withBom = false)
+    public function __construct(string $filename, string $delimiter = ',', string $enclosure = '"', string $escape = '\\', bool $showHeaders = true, bool $withBom = false)
```
Comment (greg0ire) [inline src/Writer/CsvWriter.php]: Told you things would get ugly when adding type hinting :P Not using multiline signatures really grinds my gears.

**Comment by Soullivaneuh** [comment]
@greg0ire This is globally define on dev-kit. I think this could be setup later when all project will be under PHP7 requirement.

BTW, this will reduce merge conflict between branch. This is the reason why Symfony are still under long array syntax for example.

**Comment by greg0ire** [comment]
@Soullivaneuh Ok I'll revert it, since it is not a BC-break, it can wait :+1:

**Comment by core23** [comment]
Any updates here @greg0ire ?

**Comment by greg0ire** [comment]
We need to get #164 and #142 right first.

**Comment by greg0ire** [comment]
Ok it's good now. I reverted the `::class` part, let's see how this goes.

**Comment by greg0ire** [comment]
Green! Please review @sonata-project/contributors !

File: composer.json [hunk 2]
```diff
@@ -13,7 +13,7 @@
         }
     ],
     "require": {
-        "php": "^5.3 || ^7.0"
+        "php": "^7.0"
```
Comment (jordisala1991) [inline composer.json]: Make it up to PHP 7.1 please, to match the other bundles composer.json

File: src/Source/DoctrineODMQuerySourceIterator.php [hunk 3]
```diff
@@ -125,15 +125,15 @@ public function rewind()
     /**
      * @param string $dateTimeFormat
      */
-    public function setDateTimeFormat($dateTimeFormat)
+    public function setDateTimeFormat(string $dateTimeFormat)
```
Comment (core23) [inline src/Source/DoctrineODMQuerySourceIterator.php]: `: void`?

File: src/Source/DoctrineORMQuerySourceIterator.php [hunk 4]
```diff
@@ -133,15 +133,15 @@ public function rewind()
     /**
      * @param string $dateTimeFormat
      */
-    public function setDateTimeFormat($dateTimeFormat)
+    public function setDateTimeFormat(string $dateTimeFormat)
```
Comment (core23) [inline src/Source/DoctrineORMQuerySourceIterator.php]: `: void`?

File: src/Source/PropelCollectionSourceIterator.php [hunk 5]
```diff
@@ -127,15 +127,15 @@ public function rewind()
     /**
      * @param string $dateTimeFormat
      */
-    public function setDateTimeFormat($dateTimeFormat)
+    public function setDateTimeFormat(string $dateTimeFormat)
```
Comment (core23) [inline src/Source/PropelCollectionSourceIterator.php]: `: void`?

**Comment by greg0ire** [comment]
@sonata-project/contributors please merge?

File: src/Source/XmlExcelSourceIterator.php [hunk 6]
```diff
@@ -50,7 +50,7 @@ public function tagStart($parser, $name, $attributes = array())
     /**
      * {@inheritdoc}
      */
-    public function tagEnd($parser, $name)
+    public function tagEnd($parser, string $name)
```
Comment (core23) [inline src/Source/XmlExcelSourceIterator.php]: return void type?

File: src/Source/XmlExcelSourceIterator.php [hunk 7]
```diff
@@ -71,7 +71,7 @@ public function tagEnd($parser, $name)
     /**
      * {@inheritdoc}
      */
-    public function tagContent($parser, $data)
+    public function tagContent($parser, string $data)
```
Comment (core23) [inline src/Source/XmlExcelSourceIterator.php]: return void type?

File: src/Writer/CsvWriter.php (hunk 1 above)
Comment (core23) [reply src/Writer/CsvWriter.php]: Maybe we should find an other way if we have more than 4(?) parameters for a method

File: src/Writer/XmlWriter.php [hunk 8]
```diff
@@ -112,9 +112,9 @@ public function write(array $data)
 
     /**
      * @param string $name
-     * @param string $value
+     * @param mixed  $value
      */
-    protected function generateNode($name, $value)
+    protected function generateNode(string $name, $value)
```
Comment (core23) [inline src/Writer/XmlWriter.php]: return void type?

File: src/Writer/CsvWriter.php (hunk 1 above)
Comment (greg0ire) [reply src/Writer/CsvWriter.php]: An option resolver might be it

**Comment by greg0ire** [comment]
please review again @core23

**Comment by OskarStark** [comment]
Thank you @greg0ire 👍

**Pull request merged**